IdentityCore/src/controllers/MSIDRequestControllerFactory.m

```diff
@@ -101,6 +130,86 @@ @implementation MSIDRequestControllerFactory
     return localController;
 }
 
++ (nullable id<MSIDRequestControlling>)v2SilentControllerForParameters:(MSIDRequestParameters *)parameters
+                                                        forceRefresh:(BOOL)forceRefresh
+                                                         skipLocalRt:(MSIDSilentControllerLocalRtUsageType)skipLocalRt
+                                                tokenRequestProvider:(id<MSIDTokenRequestProviding>)tokenRequestProvider
+                                                               error:(NSError *__autoreleasing*)error
+{
+    // Nested auth protocol - Reverse client id & redirect uri
+    if ([parameters isNestedAuthProtocol])
+    {
+        [parameters reverseNestedAuthParametersIfNeeded];
+    }
+    
+    MSIDSilentController *fallbackController = nil;
+    
+    if ([parameters shouldUseBroker])
+    {
+        if (parameters.allowUsingLocalCachedRtWhenSsoExtFailed)
+        {
+            fallbackController = [[MSIDSilentController alloc] initWithRequestParameters:parameters
+                                                                         forceRefresh:YES
+                                                                 tokenRequestProvider:tokenRequestProvider
+                                                                                error:error];
+            fallbackController.isLocalFallbackMode = YES;
+        }
+    
+        MSIDSilentController *xpcController = nil;
+#if TARGET_OS_OSX
+        if (parameters.msidXpcMode != MSIDXpcModeDisable && [MSIDXpcSilentTokenRequestController canPerformRequest])
+        {
+            xpcController = [[MSIDXpcSilentTokenRequestController alloc] initWithRequestParameters:parameters
+                                                                                           forceRefresh:forceRefresh
+                                                                                   tokenRequestProvider:tokenRequestProvider
+                                                                          fallbackInteractiveController:fallbackController
+                                                                                                  error:error];
+            if (parameters.msidXpcMode == MSIDXpcModeFull || parameters.msidXpcMode == MSIDXpcModeOverride)
```

Comment:
What about parameters.msidXpcMode == MSIDXpcModeBackup? In such a case, fallbackInteractiveController will be fallbackController

Reply:
Correct
in the condition `parameters.xpcMode == MSIDXpcModeFull || parameters.xpcMode == MSIDXpcModeOverride`
the `xpcController` is set to `nil`

So in `MSIDXpcModeBackup` condition, the  `xpcController` is not set to `nil`, and will be used as a `fallbackInteractiveController` when creating `MSIDSSOExtensionSilentTokenRequestController` (L188)

Sorry the `fallbackInteractiveController` is really misleading as this is silent request. I will modify the name in a different PR as it is from old code and XPC unrelated.